lib/sync-settings.coffee

```diff
@@ -36,19 +63,19 @@ module.exports =
   serialize: ->
 
   backup: (cb=null) ->
-    files =
-      "settings.json":
-        content: JSON.stringify(atom.config.settings, @filterSettings, '\t')
-      "packages.json":
-        content: JSON.stringify(@getPackages(), null, '\t')
-      "keymap.cson":
-        content: (@fileContent atom.keymaps.getUserKeymapPath()) ? "# keymap file (not found)"
-      "styles.less":
-        content: (@fileContent atom.styles.getUserStyleSheetPath()) ? "// styles file (not found)"
-      "init.coffee":
-        content: (@fileContent atom.config.configDirPath + "/init.coffee") ? "# initialization file (not found)"
-      "snippets.cson":
-        content: (@fileContent atom.config.configDirPath + "/snippets.cson") ? "# snippets file (not found)"
+    files = {}
+    if atom.config.get('sync-settings.syncSettings')
+      files["settings.json"] = content: JSON.stringify(atom.config.settings, @filterSettings, '\t')
+    if atom.config.get('sync-settings.syncPackages')
+      files["packages.json"] = content: JSON.stringify(@getPackages(), null, '\t')
+    if atom.config.get('sync-settings.syncKeymap')
+      files["keymap.cson"] = content: (@fileContent atom.keymaps.getUserKeymapPath()) ? "# keymap file (not found)"
+    if atom.config.get('sync-settings.syncStyles')
+      files["styles.less"] = content: (@fileContent atom.styles.getUserStyleSheetPath()) ? "// styles file (not found)"
+    if atom.config.get('sync-settings.syncInit')
+      files["init.coffee"] = content: (@fileContent atom.config.configDirPath + "/init.coffee") ? "# initialization file (not found)"
+    if atom.config.get('sync-settings.syncSnippets')
+      files["snippets.cson"] = content: (@fileContent atom.config.configDirPath + "/snippets.cson") ? "# snippets file (not found)"
```

Comment:
Can we replace with a map like:

```coffeescript
fileConfig = 
  'settings.json': => JSON.stringify(atom.config.settings, @filterSettings, '\t')
```

and then just iterate over it like
```coffeescript
files = for file, reader of fileConfig
  reader() if atom.config.get("sync-settings.sync#{_.capitalize file}")
```


The map could be extended to include also a writer, and then it can be used in both backup and restore.